Fix: Handle timestamps correctly and update documentation types

andyleap opened the pull request:
This commit addresses two issues:

1. **Timestamp handling:** Planned values weren't working correctly during creation, so timestamp handling was reworked, and basic tests to ensure it's handled correctly were added.

2. **Documentation Type Correction:** The documentation incorrectly stated that the `number` field in the resource definition accepted a string. This has been corrected to reflect the actual type, which is an integer.

andyleap committed:
Fix: Handle timestamps correctly and update documentation types

This commit addresses two issues:

1. **Timestamp handling:** Planned values weren't working correctly during creation, so timestamp handling was reworked, and basic tests to ensure it's handled correctly were added.

2. **Documentation Type Correction:** The documentation incorrectly stated that the `number` field in the resource definition accepted a string. This has been corrected to reflect the actual type, which is an integer.

austinpray-mixpanel reviewed (approved):
lgtm % make linter happy

andyleap committed:
Refactor: Remove debug logging

This commit removes the `testlog` function, which was previously used for debugging purposes.

The debug logging is no longer needed and has been removed to improve code clarity.

andyleap committed:
Fix go.mod

andyleap committed:
feat: Allow user-provided timestamps for MultiRotateSet

This commit introduces the ability for users to provide their own timestamps when creating MultiRotateSet resources. Previously, the timestamp was always automatically generated.

This enhancement provides more flexibility, allowing users to:

- Align rotations with specific time periods.
- Backfill historical data with defined timestamps.

The logic now checks if a timestamp is explicitly provided. If so, it attempts to parse it according to RFC3339 format. If parsing fails, an error is returned. Otherwise, the provided timestamp is used. If no timestamp is provided, the current time is used, maintaining the existing default behavior.
